shared/modules/add-nonce-to-csp.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Adds a nonce value to the script-src directive of the Content-Security-Policy Header.
+ *
+ * @param header - the Content-Security-Policy Header
+ * @param nonce - the nonce value to add
+ * @returns the Content-Security-Policy Header with the nonce value added
+ */
+export const addNonceToCsp = (header: string, nonce: string) => {
+  return header.replace(
+    /(^|;[\t\n\f\r ]*)script-src([^;]*)/iu,
```

Comment:
What would happen if the entire policy is just `default-src 'none'`, or something equally restrictive? I _think_ we'd still be blocked in that case, as `default-src` is a fallback for absent directives. We might need to add some more logic here.

---

We _might_ need to expand on this even further by expanding/adding directives to allow for `img-src data:;` in order for our provider's `icon` to be allowed. Wallet connectors can use this icon when providing the user with their connection options.

Example of the icon in use from [our test-dapp](https://metamask.github.io/test-dapp/):

![image](https://github.com/user-attachments/assets/dc2b5ff3-8543-422d-81e9-1f1c2d74d140)

I haven't tested if this CSP policy is broken in FF or not. Actually, I haven't tested if this is already broken in Chrome either!

Reply:
Note: I still worry about potential regex backtracking and self-inflicted DoS.